cmake/SetupCompilers.cmake

```diff
@@ -59,6 +59,9 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
 elseif (CMAKE_CXX_COMPILER_ID MATCHES AppleClang)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  set(CMAKE_CXX_STANDARD 11) # C++11...
+  set(CMAKE_CXX_STANDARD_REQUIRED ON) #...is required...
+  set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
@trws would adding `set(CMAKE_CXX_EXTENSIONS OFF)` to our default at least prevent `-std=gnu++1[14]` from being emitted? That way, if the compiler supports C++14 it will use it but still decay to C++11?